Simplified/Reader2/LCP/LCPLibraryService.swift

```diff
@@ -0,0 +1,151 @@
+//
+//  LCPLibraryService.swift
+//  r2-testapp-swift
+//
+//  Created by Mickaël Menu on 01.02.19.
+//
+//  Copyright 2019 Readium Foundation. All rights reserved.
+//  Use of this source code is governed by a BSD-style license which is detailed
+//  in the LICENSE file present in the project repository where this source code is maintained.
+//
+
+#if LCP
+
+import Foundation
+import UIKit
+import R2Shared
+import ReadiumLCP
+
+
+@objc class LCPLibraryService: NSObject, DRMLibraryService {
+  
+  private var lcpService =  R2MakeLCPService()
+  
+  /// [LicenseDocument.id: passphrase callback]
+  private var authenticationCallbacks: [String: (String?) -> Void] = [:]
+  
+  /// Library service brand
+  let brand: DRM.Brand = .lcp
+  
+  /// Returns whether this DRM can fulfill the given file into a protected publication.
+  /// - Parameter file: file URL
+  /// - Returns: `true` if file contains LCP DRM license information.
+  func canFulfill(_ file: URL) -> Bool {
+    return file.pathExtension.lowercased() == "lcpl"
+  }
+  
+  /// Fulfill LCP license publication
+  /// This function was added for compatibility with Objective-C NYPLMyBooksDownloadCenter.
+  /// - Parameters:
+  ///   - file: LCP license file.
+  ///   - completion: Complition is called after a publication was downloaded or an error received.
+  ///   - localUrl: Downloaded publication URL.
+  ///   - downloadTask: `URLSessionDownloadTask` that downloaded the publication.
+  ///   - error: `NSError` if any.
+  @objc func fulfill(_ file: URL, completion: @escaping (_ localUrl: URL?, _ error: NSError?) -> Void) {
+    lcpService.importPublication(from: file, authentication: self) { result, error in
+      var nsError: NSError?
+      if let error = error {
+        let domain = "LCP fulfillment error"
+        let code = 0
```

Comment:
I would avoid creating this new error instance since it's not really adding much. I assume that "LCP fulfillment error" is for logging, and for that it's better to specify it on the call to NYPLErrorLogger (see other comment).

Here I'd just call completion relaying the `error` you received directly.

Reply:
We can't do that because LCPError can't be passed to Objective-C code

Reply:
Maybe I'm wrong, but I assumed `error` was typed as `Error` and so it could be safely converted to NSError by the compiler/runtime. That way you'd preserve the original `error` object without losing data, and put the "LCP fulfillment error" where it's actually used (i.e. for NYPLErrorLogger).